lib/replicate/checkpointer.js

```diff
@@ -5,24 +5,60 @@ var explain404 = require('./../deps/explain404');
 var pouchCollate = require('pouchdb-collate');
 var collate = pouchCollate.collate;
 
-function updateCheckpoint(db, id, checkpoint, returnValue) {
+var REPLICATION_ID_VERSION = 2;
+// This is an arbitrary number to limit the
+// amount of replication history we save in the checkpoint.
+// If we save too much, the checkpoing docs will become very big,
+// if we save fewer, we'll run a greater risk of having to
+// read all the changes from 0 when checkpoint PUTs fail
+// CouchDB 2.0 has a more involved history pruning,
+// but let's go for the simple version for now.
+var CHECKPOINT_HISTORY_SIZE = 5;
```

Comment:
CouchDB keeps history size up to [50 records](https://github.com/apache/couchdb-couch-replicator/blob/master/src/couch_replicator.erl#L762-L766). 5 is too small value especially assuming that replication ID is based on remote host URL (if it's really true IDv2) which has higher collision risks.